doc/source/reference/tokenizers/token_ngram.rst

```diff
@@ -0,0 +1,194 @@
+.. -*- rst -*-
+
+.. highlightlang:: none
+
+.. groonga-command
+.. database: tokenizers
+
+.. _token-ngram:
+
+``TokenNgram``
+==============
+
+Summary
+-------
+
+``TokenNgram`` can define its behavior dynamically via its options.
+For example, we can use it as unigram, bigram, trigram on with changing ``n`` option value as below.
+
+Uni-gram:
+
+.. groonga-command
+.. include:: ../../example/reference/tokenizers/token-ngram-uni-gram.log
+.. tokenize --tokenizer 'TokenNgram("n", 1)' --string "Hello World"
+
+Bi-gram:
+
+.. groonga-command
+.. include:: ../../example/reference/tokenizers/token-ngram-bi-gram.log
+.. tokenize --tokenizer 'TokenNgram("n", 2)' --string "Hello World"
+
+tri-gram:
+
+.. groonga-command
+.. include:: ../../example/reference/tokenizers/token-ngram-tri-gram.log
+.. tokenize --tokenizer 'TokenNgram("n", 3)' --string "Hello World"
+
+This tokenizer also has options other than the above.
+
+Syntax
+------
+
+``TokenNgram`` has optional parameter.
+
+No options::
+
+  TokenNgram
+
+If we don't use options, ``TokenNgram`` is the same behavior for ``TokenBigram``.
+
+Specify option::
+
+  TokenNgram("n", true)
+
+  TokenNgram("loose_symbol", true)
+
+  TokenNgram("loose_blank", true)
+
+  TokenNgram("remove_blank", true)
+
+  TokenNgram("report_source_location", true)
+
+  TokenNgram("unify_alphabet", true)
+
+  TokenNgram("unify_symbol", true)
+
+  TokenNgram("unify_digit", true)
+
+Specify multiple options::
+
+  TokenNgram("loose_symbol", true, "loose_blank", true)
+
+``TokenNgram`` also specify multiple options as above.
+We can also specify mingle multiple options except the above example.
```

Comment:
It is simple to use `We can also specify multiple options at the same time except the above examples.`